### PR TITLE
Fix spancat for zero suggestions

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

`spancat.predict` crashes on docs with zero suggestions.

### Types of change

Bug fix.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
To clarify: I only created this PR to show the failing test. I don't have concrete plans to work on this immediately. Anyone else is welcome to work it.

File: spacy/ml/extract_spans.py [hunk 1]
```diff
@@ -32,8 +32,8 @@ def forward(
         Y = Ragged(X.dataXd[indices], spans.dataXd[:, 1] - spans.dataXd[:, 0])  # type: ignore[arg-type, index]
     else:
         Y = Ragged(
-            ops.xp.zeros(X.dataXd.shape, dtype=X.dataXd.dtype),
-            ops.xp.zeros((len(X.lengths),), dtype="i"),
+            ops.xp.zeros((0, 0), dtype=X.dataXd.dtype),
+            ops.xp.zeros((0,), dtype="i"),
```
Comment (adrianeboyd) [inline spacy/ml/extract_spans.py:36]: This is now never reached in `spancat`. The previous placeholder was clearly incorrect. This shouldn't crash the model now, but it's also not really necessary. You could just require that `len(indices) > 0` instead.

Comment (adrianeboyd) [reply spacy/ml/extract_spans.py]: Nevermind about this, this was wrong.

File: spacy/tests/pipeline/test_spancat.py [hunk 2]
```diff
@@ -372,31 +372,56 @@ def test_overfitting_IO_overlapping():
...
-
-        return zero_suggester
+            spans = []
+            lengths = []
+            for doc in docs:
+                if len(doc) > 0 and len(doc) % 2 == 0:
+                    spans.append((0, 1))
+                    lengths.append(1)
+                else:
+                    lengths.append(0)
```
Comment (polm) [inline spacy/tests/pipeline/test_spancat.py:389]: The test here is passing, but I'm having trouble understanding why having the `spans` and `lengths` being different lengths works. Is it intentional that they end up different lengths? Should a Ragged where the input array and the lengths are different lengths be valid?

Comment (adrianeboyd) [reply spacy/tests/pipeline/test_spancat.py:389]: `lengths` is the number of spans per doc. Here, some docs have 1 span and some docs have 0 spans.

Comment (polm) [reply spacy/tests/pipeline/test_spancat.py:389]: Got it, that makes sense. I had some trouble following the difference between span lengths, number of spans per doc, and indices in used in extracting spans.

**Review by polm — APPROVED** [review]
Looks good.

**Pull request merged**